nimrod/report/report_analysis.py

```diff
@@ -4,110 +4,103 @@
 
 class Report_Analysis:
 
-    def __init__(self, randoop_original, randoop_modified):
-        self.randoop_original = randoop_original
-        self.randoop_modified = randoop_modified
-
-    def start_analysis(self):
-        left_randoop_original = self.randoop_original[0][2]
-        left_randoop_modified = self.randoop_modified[0][2]
-
-        if os.path.isdir(left_randoop_original) and os.path.isdir(left_randoop_modified):
-            self.suites_comparison(left_randoop_original, left_randoop_modified)
-            try:
-                right_randoop_original = self.randoop_original[4][2]
-                right_randoop_modified = self.randoop_modified[4][2]
-                self.suites_comparison(right_randoop_original, right_randoop_modified)
-            except:
-                print("It wasn't possible to compare the test suites for the commit: " + str(self.randoop_original[1][4]))
+    def start_analysis(self, randoop_original, randoop_modified):
+        self.suites_comparison(randoop_original[0][2], randoop_modified[0][2])
+        right_path_index_original = len(randoop_original) - 1
+        right_path_index_modified = len(randoop_modified) - 1
+        self.suites_comparison(randoop_original[right_path_index_original][2], randoop_modified[right_path_index_modified][2])
+
+    def checking_suites_and_reports(self, path_suite_original, path_suite_modified):
+        if os.path.isdir(path_suite_original) and os.path.isdir(path_suite_modified):
+            if self.all_reports():
+                self.suites_comparison(path_suite_original, path_suite_modified)
+            else:
+                print(
+                    "It wasn't possible to compare the test suites because there are missing reports in paths: " + str(
+                        path_suite_original) + " or " + str(path_suite_modified))
         else:
-            print("It wasn't possible to compare the test suites for commits: " + str(self.randoop_original[0][4]) + " and " + str(self.randoop_original[1][4]))
+            print("It wasn't possible to make analysis because there are invalid path suites")
+
+    def all_reports(self):
+        return os.path.isfile(path_suite_original + "/methods_report.csv") and os.path.isfile(path_suite_modified + "/methods_report.csv") \
+               and os.path.isfile(path_suite_original + "/objects_report.csv") and os.path.isfile(path_suite_modified + "/objects_report.csv")
 
     def suites_comparison(self, path_suite_original, path_suite_modified):
         methods = self.methods_report_comparison(path_suite_original, path_suite_modified)
         objects = self.objects_report_comparison(path_suite_original, path_suite_modified)
 
         if len(methods) == 0 or len(objects) == 0:
-            print("It wasn't possible to compare the suites " + str(path_suite_original) + " and " + str(path_suite_modified))
+            print("It wasn't possible to compare the suites " + str(path_suite_original) + " and " + str(
```

Comment:
In our actual implementation, I don't believe that just one report will be generated. Except for any error, the makes randoop crush during he was writing the second report, in this occasional error, we will have just one report. In theory, there will be reports even if those are just empty files - but I don't believe this will happen, because for this we must have a test suite that never calls any methods or never create any object. We can change this if you think it is better.